server/server.go

```diff
@@ -0,0 +1,80 @@
+// Package server implements the core HTTP server
+package server
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/flashbots/go-template/config"
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+type Server struct {
+	cfg         *config.Server
+	id          uuid.UUID
+	isHealthy   bool
+	isHealthyMx sync.RWMutex
+	log         *zap.Logger
+	srv         *http.Server
+}
+
+func New(cfg *config.Server) *Server {
+	id := uuid.Must(uuid.NewRandom())
+	s := &Server{
+		cfg:         cfg,
+		id:          id,
+		isHealthy:   true,
+		isHealthyMx: sync.RWMutex{},
+		log:         cfg.Log.With(zap.String("serverID", id.String())),
```

Comment:
- can we reduce the noise by md5'ing the uid?
- let's add version to the default log too